src/SFA.DAS.AssessorService.Data/RegisterRepository.cs

```diff
@@ -77,29 +77,71 @@ public async Task<string>CreateEpaOrganisationStandard(EpaOrganisationStandard o
+                        new
+                        {
+                            organisationStandard.OrganisationId,
+                            organisationStandard.StandardCode,
+                            organisationStandard.EffectiveFrom,
+                            organisationStandard.EffectiveTo,
+                            organisationStandard.DateStandardApprovedOnRegister,
+                            organisationStandard.Comments,
+                            organisationStandard.ContactId,
+                            organisationStandard.OrganisationStandardData,
+                            organisationStandard.StandardReference
+                        }).Single();
+
+                    foreach (var deliveryAreaId in deliveryAreas.Distinct())
                     {
-                        organisationStandard.OrganisationId, organisationStandard.StandardCode,
-                        organisationStandard.EffectiveFrom, organisationStandard.EffectiveTo,
-                        organisationStandard.DateStandardApprovedOnRegister, organisationStandard.Comments,
-                        organisationStandard.ContactId, organisationStandard.OrganisationStandardData
-                    }).Single();
+                        connection.Execute("INSERT INTO OrganisationStandardDeliveryArea ([OrganisationStandardId],DeliveryAreaId, Status) VALUES " +
+                                            "(@osdaId, @deliveryAreaId,'Live'); ",
+                                        new { orgStandardId, deliveryAreaId }
+                                        );
+                    }
+                }
+                else
+                {
+                    // Fix StandardReference on the existing record
+                    connection.Execute("UPDATE [dbo].[OrganisationStandard] SET StandardReference = @standardReference WHERE Id = @id",
+                        new 
+                        {
+                            standardReference = organisationStandard.StandardReference,
+                            id = orgStandardId
+                        });
+                }
 
-                foreach (var deliveryAreaId in deliveryAreas.Distinct())
+                if(null != organisationStandard.StandardVersions)
                 {
-                    connection.Execute("INSERT INTO OrganisationStandardDeliveryArea ([OrganisationStandardId],DeliveryAreaId, Status) VALUES " + 
-                                        "(@osdaId, @deliveryAreaId,'Live'); ",
-                                    new { osdaId, deliveryAreaId}
-                                    );
-                }                      
-                        
-                return osdaId;
-            }
-            
+                    foreach (var version in organisationStandard.StandardVersions)
```

Comment:
Like above, if it already exists it falls into an update, do we want to do a check here if the org standard version exists ( if we don't update the other validation rule called out in a comment ) otherwise we get a key constraint error.

I had an issue whereby this page errored, and i had to refresh then it kept falling over as it wouldn't get past the key constraint